### ISSUE TITLE
DPE Inspector: Modifying one of three Vegetation System Settings component's properties causes Editor to crash

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16702]
**Describe the bug**
If `ed_enableDPE` is `true` then modifying *View Area Grid Size*, *Sector Point Density*, or *Sector Size In Meters* property in the *Vegetation System Settings* component causes the Editor to crash.

Call Stack:
```
Call Stack Trace:
25) 00007FFDE3D4DEA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFDE3586A06 (InProcessClient64) : (function-name not available)
23) 00007FFDE3586A98 (InProcessClient64) : (function-name not available)
22) 00007FFDE3586961 (InProcessClient64) : (function-name not available)
21) 00007FFDE6395330 (ntdll) : memset
20) 00007FFDE637C876 (ntdll) : _C_specific_handler
19) 00007FFDE639221F (ntdll) : _chkstk
18) 00007FFDE63414B4 (ntdll) : RtlRaiseException
17) 00007FFDE6390D2E (ntdll) : KiUserExceptionDispatcher
16) C:\o3de\Gems\Vegetation\Code\Source\AreaSystemComponent.cpp (253) : Vegetation::AreaSystemConfig::ValidateSectorSize
15) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (661) : AZ::AttributeMemberFunction<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > (__cdecl Vegetation::AreaSystemConfig::*)(void *,AZ::Uuid const &)>::Inv
14) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,AZ::Uuid const &),AZStd::Inter
13) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (685) : AZStd::function<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,AZ::Uuid const &)>::operator()
12) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (468) : AZ::InvokeFromDomArray<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> >,void *,AZ::Uuid const &>
11) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (686) : AZ::AttributeMemberFunction<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > (__cdecl Vegetation::AreaSystemConfig::*)(void *,AZ::Uuid const &)>::Dom
10) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentSchema.h (557) : AZ::DocumentPropertyEditor::CallbackAttributeDefinition<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,A
 9) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentSchema.h (614) : AZ::DocumentPropertyEditor::CallbackAttributeDefinition<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,A
 8) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyEditorAPI_Internals.h (491) : AzToolsFramework::RpePropertyHandlerWrapper<int>::OnValueChanged
 7) C:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (807) : AZ::Internal::EBusContainer<AzToolsFramework::IndividualPropertyHandlerEditNotifications,AzToolsFramework::IndividualPropertyHandlerEditNotifications,1,0>::Dispatcher<AZ::EBus<AzTo
 6) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyManagerComponent.cpp (204) : AzToolsFramework::Components::PropertyManagerComponent::RequestWrite
 5) C:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzToolsFramework::PropertyEditorGUIMessages,AzToolsFramework::PropertyEditorGUIMessages,0,1>::Dispatcher<AZ::EBus<AzToolsFramework::PropertyEditorGUIMe
 4) C:\Users\GPC-589\.o3de\3rdParty\packages\qt-5.15.2-rev7-windows\qt\include\QtCore\qobjectdefs_impl.h (449) : QtPrivate::QFunctorSlotObject<`AzToolsFramework::IntWidgetHandler<unsigned int,AzToolsFramework::PropertyIntSpinCtrl,AzToolsFramework::IntSpinBox
 3) 00007FFDADD309AA (Qt5Core) : QObject::qt_static_metacall
 2) C:\o3de\AutomatedTesting\build\windows\o3de\Code\Framework\AzToolsFramework\AzToolsFramework_autogen\include_profile\UI\PropertyEditor\moc_PropertyIntSpinCtrl.cpp (177) : AzToolsFramework::PropertyIntSpinCtrl::qt_static_metacall
 1) 00007FFDADD30A08 (Qt5Core) : QObject::qt_static_metacall
```

This issue occurs on both Windows and Linux OS.

This issue does not occur with `ed_enableDPE` set to `false`.

**Steps to reproduce**
1. Execute `ed_enableDPE 1` command in the Console.
2. Select the level prefab in the *Entity Outliner*.
3. Add a *Vegetation System Settings* component in the *Inspector*.
4. Modify *View Area Grid Size*, *Sector Point Density*, or *Sector Size In Meters* property on the *Vegetation System Settings*.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes.

**Video and all crash related files**

https://github.com/o3de/o3de/assets/86952082/786d71ad-3430-4819-b95c-e2e24b35ef90

[Editor.log](https://github.com/o3de/o3de/files/12596133/Editor.log)
[error.log](https://github.com/o3de/o3de/files/12596135/error.log)
[error.zip](https://github.com/o3de/o3de/files/12596136/error.zip) - Contains `error.dmp` file.
[CallStackVS.txt](https://github.com/o3de/o3de/files/12596141/CallStackVS.txt)

Visual Studio crash screenshot: 
![VS](https://github.com/o3de/o3de/assets/86952082/7e919f0e-8cdc-47ce-a3a3-fcbe216365d5)

**Found in Branch**
Stabilization/2310 (6f4ea1d)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
6f4ea1dbf9f084abdfdfabd3f43109eab2142188

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.3 LTS
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by alexmontAmazon** [comment]
should be fixed by https://github.com/o3de/o3de/pull/16732

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (9b90a24) branch.
